src/session_impl.cpp

```diff
@@ -174,14 +174,27 @@ namespace
 #include <winerror.h>
 #endif
 
-using libtorrent::aux::session_impl;
 using namespace std::placeholders;
 
 #ifdef BOOST_NO_EXCEPTIONS
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-noreturn"
+#endif
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wmissing-noreturn"
+#endif
 namespace boost {
-	void throw_exception(std::exception const& e) { ::abort(); }
+	void throw_exception(std::exception const&) { ::abort(); }
```

Comment:
yeah, and the reason boost and asio doesn't define it as ``abort()`` or ``terminate()`` is because it's up to the application to decide what to do. it seems like it would make more sense for the client to decide, rather than libtorrent